elfloader-tool/src/drivers/driver.c

```diff
@@ -53,3 +53,32 @@ int initialise_devices(void)
 
     return 0;
 }
+
+static int init_device_non_boot(struct elfloader_device *dev)
+{
+    struct elfloader_driver **drvp = __start__driver_list;
+
+    while (drvp < __stop__driver_list) {
+        struct elfloader_driver *drv = *drvp;
+        if (drv->init_non_boot) {
+            int ret = table_has_match(dev->compat, drv->match_table);
+            if (ret >= 0) {
+                dev->drv = drv;
+                drv->init_non_boot(dev, drv->match_table[ret].match_data);
```

Comment:
The return value of `drv->init_non_boot()` should be propagated, otherwise my old comment stands.